ext/src/ruby_api/store.rs

```diff
@@ -113,6 +115,52 @@ impl Store {
     }
 }
 
+/// A wrapper around a Ruby Value that has a store context.
+/// Used in places where both Store or Caller can be used.
+#[derive(Debug)]
+pub enum StoreContextValue<'a> {
+    Store(WrappedStruct<Store>),
+    Caller(WrappedStruct<Caller<'a>>),
+}
```

Comment:
After rebasing on #14, I switched this to use `WrappedStruct`.

Pros:
- Cant' fail to convert, I went from `TryFrom<Value> for StoreContextValue` to `From<WrappedStruct<Store>> for StoreContextValue`.
- We only need to convert once.

Cons: `Func`, `Memory`, etc. all have to know about the lifetime. Which also means we can't derive `DataType` and have to use the more verbose `DataTypeBuilder`.

I think I like this better. But it's an easy thing to change, so 🤷.